Jenkinsfile

```diff
@@ -0,0 +1,197 @@
+        // workaround for now to speed-up the build: only build the product on develop and master
+        def buildProduct = env.BRANCH_NAME == 'develop' || isMaster() ? '-Pproduct' : ''
+        def goal = isMaster() ? 'deploy' : 'install'
+        withXvfb {
+            mvn "clean $goal $buildProduct -Dmaven.test.failure.ignore -Dsurefire.useFile=false -Dtycho.localArtifacts=ignore"
+        }
+    }
+
+    if (isMaster()) {
+        postRelease(preReleaseVersion)
+    }
+
+    stage 'Collect test results'
+    step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
+    codecov('codecov_test-editor-xtext')
+}
+
+boolean isMaster() {
+    return env.BRANCH_NAME == 'master'
+}
+
+void prepareRelease() {
+    stage 'Prepare release'
+
+    // Remove SNAPSHOT version
+    echo 'Removing SNAPSHOT from target platform version'
+    def String noSnapshotVersion = '\\${parsedVersion.majorVersion}.\\${parsedVersion.minorVersion}.\\${parsedVersion.incrementalVersion}'
+    setVersion(noSnapshotVersion, 'releng/org.testeditor.releng.target/pom.xml', 'org.testeditor.releng.target.parent')
+    echo 'Removing SNAPSHOT from test-editor version'
+    setVersion(noSnapshotVersion, 'pom.xml', 'org.testeditor.releng.parent')
+
+    // Set the display name for the job to the version
+    String version = getCurrentVersion()
+    currentBuild.displayName = version
+    echo "Version to release is: $version"
+}
+
+String getCurrentVersion() {
+    def pom = readMavenPom file: 'pom.xml'
+    return pom.parent.version
+}
+
+String getGitUrl() {
+    return scm.userRemoteConfigs.first().url
+}
+
+String getGitUrlAsSsh() {
+    return getGitUrl().replace("https://github.com/", "git@github.com:")
+}
+
+void postRelease(String preReleaseVersion) {
+    stage 'Tag release'
+
+        def version = "v${getCurrentVersion()}"
+        echo "Tagging release as $version"
+        sh "git add *"
+        sh "git commit -m '[release] $version'"
+        sh "git tag $version"
+        // workaround: cannot push without credentials using HTTPS => push using SSH
+        sh "git remote set-url origin ${getGitUrlAsSsh()}"
+        sh "git push origin master --tags"
+
+    stage 'Increment develop version'
+        sh "git checkout develop"
+        sh "git reset --hard origin/develop"
+        def developVersion = getCurrentVersion()
+        if (developVersion == preReleaseVersion) {
+            sh "git merge origin/master"
+            def nextSnapshotVersion = '\\${parsedVersion.majorVersion}.\\${parsedVersion.minorVersion}.\\${parsedVersion.nextIncrementalVersion}-SNAPSHOT'
+            setVersion(nextSnapshotVersion, 'releng/org.testeditor.releng.target/pom.xml', 'org.testeditor.releng.target.parent')
+            setVersion(nextSnapshotVersion, 'pom.xml', 'org.testeditor.releng.parent')
+            sh "git add *"
+            sh "git commit -m '[release] set version ${getCurrentVersion()}'"
```

Comment:
The closure `${getCurrentVersion()}` retrieves the current version after the `setVersion` was executed. `nextSnapshotVersion` is only a pattern that is passed to Maven and gets interpreted. The script doesn't now the next version yet, so it needs to read it from the POM.